### PR TITLE
generate event_consts

### PR DESCRIPTION
**Pull request by OlexandrNazin**

### CODE REVIEW

**Comment by OlexandrNazin** [comment]
@taaraora , это правильное решение?

**Comment by OlexandrNazin** [comment]
@taaraora, this is the right decision?

File: render_event_consts.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+	for _, v := range renderer.Model.ComponentsSpecificationExtension {
+		if v.Value.Yaml != "" {
+
+			constants := strings.Split(v.Value.Yaml, "\n")
+			f.WriteLine(`const (`)
+			for i := 0; i < (len(constants) - 1); i++ {
+
+				f.WriteLine(constants[i][2:] + `FactStreamType` + `=` + `"` + constants[i][2:] + `_facts"`)
+
```
Comment (taaraora) [inline render_event_consts.go:63]: do we really need this empty line?

File: render_event_consts.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+func (renderer *Renderer) Render_event_consts() ([]byte, error) {
+	f := NewLineWriter()
+	f.WriteLine(`// GENERATED FILE: DO NOT EDIT!`)
+	f.WriteLine(``)
+	f.WriteLine(`package ` + renderer.Package)
+
+	for _, v := range renderer.Model.ComponentsSpecificationExtension {
+		if v.Value.Yaml != "" {
+
+			constants := strings.Split(v.Value.Yaml, "\n")
```
Comment (taaraora) [inline render_event_consts.go]: Could you please make this logic (L12, L14, L16, L18) as a separate function and add there some unit tests for it.

**Review by taaraora — CHANGES_REQUESTED** [review]

**Comment by OlexandrNazin** [comment]
@taaraora made edits

File: render_event_constants_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	}{
+		{
+			A:      "- users_commands\n- accounts_commands\n- idp_commands\n",
+			Result: []string{"users_commands", "accounts_commands", "idp_commands"},
+		},
+		{
+			A:      "- users_facts\n- accounts_facts\n- idp_facts\n",
+			Result: []string{"users_facts", "accounts_facts", "idp_facts"},
+		},
+	}
```
Comment (taaraora) [inline render_event_constants_test.go:30]: could you please add a few negative cases.

File: render_event_consts.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	f.WriteLine(`const (`)
+	//generate constants CommandsStreamType
+	if len(constants) >= 1 {
+		for _, constant := range splitLineConstants(constants[1]) {
+			f.WriteLine(constant + `CommandsStreamType` + `=` + `"` + constant + `_commands"`)
+		}
+	}
+	f.WriteLine(`)`)
+	return f.Bytes(), nil
+}
```
Comment (taaraora) [inline render_event_consts.go:110]: please `go fmt` your files

**Review by taaraora — CHANGES_REQUESTED** [review]

File: render_event_consts.go (hunk 4 above)
Comment (OlexandrNazin) [reply render_event_consts.go:110]: it was done, maybe it doesn't work as you would like.
What needs to be fixed?
Added two blank lines.

File: render_event_constants_test.go [hunk 5]
```diff
@@ -1,32 +1,54 @@
...
 func Test_splitLineConstants(t *testing.T) {
 	testData := []struct {
 		A      string
+		name   string
 		Result []string
+		err    error
 	}{
 		{
 			A:      "- users_commands\n- accounts_commands\n- idp_commands\n",
+			name:   "",
```
Comment (taaraora) [inline render_event_constants_test.go]: let's add meaningful names for TCs and rename `A` filed to `input`.

File: render_event_constants_test.go [hunk 6]
```diff
@@ -1,32 +1,54 @@
...
 			Result: []string{"users_commands", "accounts_commands", "idp_commands"},
+			err:    error(nil),
```
Comment (taaraora) [inline render_event_constants_test.go]: here just use `nil` value

File: render_event_constants_test.go [hunk 7]
```diff
@@ -1,32 +1,54 @@
...
+			name:   "",
 			Result: []string{"users_facts", "accounts_facts", "idp_facts"},
+			err:    error(nil),
 		},
 		{
-			A:      "- users_test\n",
-			Result: []string{"users_test"},
+			A:      "properties:\n  code:\n    type: integer\n    format: int64\n    message:\n    type: string\n",
+			name:   "ls",
+			Result: []string{"properties:", "  code:", "    type: integer", "    format: int64", "    message:", "    type: string"},
```
Comment (taaraora) [inline render_event_constants_test.go]: when you got error you can keep value empty.

File: render_event_constants_test.go [hunk 8]
```diff
@@ -1,32 +1,54 @@
...
+		res, err := splitLineConstants(testCase.A, testCase.name)
+		if err != nil {
+
+		}
 		for i := 0; i < len(res); i++ {
 			if res[i] != testCase.Result[i] {
-				t.Fatalf("Expected %+q, got %+q\n", testCase.Result, res)
+				t.Errorf("Expected %+q, got %+q\n", testCase.Result, res)
+			}
+			if err != nil && err.Error() != testCase.err.Error() {
```
Comment (taaraora) [inline render_event_constants_test.go]: it is better to check TC's error and if it is nil then check values.

File: render_event_consts.go [hunk 9]
```diff
@@ -9,35 +10,57 @@ func (renderer *Renderer) Render_event_consts() ([]byte, error) {
 	f.WriteLine(`// GENERATED FILE: DO NOT EDIT!`)
 	f.WriteLine(``)
 	f.WriteLine(`package ` + renderer.Package)
-	var constants []string
+	var constants struct {
```
Comment (taaraora) [inline render_event_consts.go]: it is better to design it as 
```go
type specExtention map[string]string

var specExtentions = new(specExtention)
```

File: render_event_consts.go [hunk 10]
```diff
@@ -9,35 +10,57 @@ func (renderer *Renderer) Render_event_consts() ([]byte, error) {
...
 		}
 	}
 	f.WriteLine(`)`)
-	f.WriteLine(`const (`)
 
 	//generate constants CommandsStreamType
-	if len(constants) >= 1 {
-		for _, constant := range splitLineConstants(constants[1]) {
-			f.WriteLine(constant + `CommandsStreamType` + `=` + `"` + constant + `_commands"`)
+	if len(constants.value) >= 1 {
```
Comment (taaraora) [inline render_event_consts.go]: create to constants with needed property key and just take neede value from map.

**Review by taaraora — CHANGES_REQUESTED** [review]

**Comment by OlexandrNazin** [comment]
@taaraora tests fixed and added, test coverage increased

File: render_event_consts.go [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+			return nil, err
+		}
+		for _, v := range blockOfConstants {
+			f.WriteLine(v)
+		}
+	}
+	return f.Bytes(), nil
+}
+func generateStreamTypeConstants(yamlValue string) ([]string, error) {
+	constants, err := splitLineConstants(yamlValue)
```
Comment (taaraora) [inline render_event_consts.go]: let's move this invocation outside this function.

**Review by taaraora — APPROVED** [review]

**Pull request merged**